src/game/Unit.cpp

```diff
@@ -2099,69 +2099,91 @@ MeleeHitOutcome Unit::RollMeleeOutcomeAgainst(const Unit* pVictim, WeaponAttackT
 
     // bonus from skills is 0.04%
     int32 skillBonus  = 4 * (attackerWeaponSkill - victimMaxSkillValueForLevel);
-    int32 sum = 0;
-    int32 roll = urand(0, 10000);
-    int32 tmp = miss_chance;
+    int32 roll = urand(0, 100) * 100;
```

Comment:
Yes. Because 1% = 100. we don't need to roll percentages of 12.34, you know what I mean?